pkg/helpers/newapp/app/imageref.go

```diff
@@ -44,6 +45,14 @@ type SecretAccessor interface {
 
 type imageRefGenerator struct{}
 
+var (
+	dnsNamePattern *regexp.Regexp
+)
+
+func init() {
+	dnsNamePattern = regexp.MustCompile("[^a-zA-Z0-9-]")
+}
+
```

Comment:
```suggestion
var (
	dnsNamePattern *regexp.Regexp =  regexp.MustCompile("[^a-zA-Z0-9-]")
)

```

I don't think you need this `init` here, the regexp can just be a var, unless there is a specific reason to use `init` that I am missing?